grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractElement.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+package ai.grakn.graph.internal;
+
+import ai.grakn.concept.Concept;
+import org.apache.tinkerpop.gremlin.structure.Element;
+import org.apache.tinkerpop.gremlin.structure.Property;
+
+import static org.apache.tinkerpop.gremlin.structure.T.id;
+
+/**
+ * <p>
+ *     Graph AbstractElement
+ * </p>
+ *
+ * <p>
+ *     Base class used to represent a construct in the graph. This includes exposed constructs such as {@link Concept}
+ *     and hidden constructs such as {@link EdgeElement} and {@link Casting}
+ * </p>
+ *
+ * @author fppt
+ *
+ */
+abstract class AbstractElement<E extends Element> {
```

Comment:
I am not a fan either to be honest but `GraknElement` inside `grakn-graph` module inside `grakn` project . . . I think we get the point :) .

`AbstractElement` cannot implement `Element` because each vendor has it's own implementation so pooo